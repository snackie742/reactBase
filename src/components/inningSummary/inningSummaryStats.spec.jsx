import React from 'react';
import { shallow } from 'enzyme';
import InningSummaryStats from './inningSummaryStats';

const defaultProps = {
    className: 'testClass',
    awayValue: '3',
    homeValue: '4',
    statAbbr: 'H',
};

const setup = (propOverrides) => {
  const props = Object.assign({}, defaultProps, propOverrides);
  const wrapper = shallow(<InningSummaryStats {...props} />);
  const component = wrapper.instance();
  return { wrapper, component };
};

describe('InningSummaryStats', () => {
  describe('<InningSummaryStats/>', () => {
    describe('(Snapshot)', () => {
      test('renders as expected', () => {
        const { wrapper } = setup(defaultProps);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
