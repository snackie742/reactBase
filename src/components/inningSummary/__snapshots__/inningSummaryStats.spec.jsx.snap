// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InningSummaryStats <InningSummaryStats/> (Snapshot) renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    awayValue="3"
    className="testClass"
    homeValue="4"
    statAbbr="H"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          H
</p>,
        <p>
          3
</p>,
        <p>
          4
</p>,
      ],
      "className": "col-lg-1 scoreBoardInnings testClass",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "H",
        },
        "ref": null,
        "rendered": "H",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "3",
        },
        "ref": null,
        "rendered": "3",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "4",
        },
        "ref": null,
        "rendered": "4",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            H
</p>,
          <p>
            3
</p>,
          <p>
            4
</p>,
        ],
        "className": "col-lg-1 scoreBoardInnings testClass",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "H",
          },
          "ref": null,
          "rendered": "H",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "3",
          },
          "ref": null,
          "rendered": "3",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "4",
          },
          "ref": null,
          "rendered": "4",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
